From 57713439bc2acdd3a49d88c33d5cf90a1ed03e9f Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 9 May 2017 10:10:37 +0800
Subject: [PATCH 1769/5242] MLK-14807-1: ARM: dts: imx6qp-sabresd-hdcp: gpu
 probe failed

commit  a354a9da3f31375c45e25b4575cf6878999e3ceb from
https://source.codeaurora.org/external/imx/linux-imx.git

I2C2 have to be disabled on hdcp board since those I2C2 bus pins
used for others, that said there is no all pmic regulators. In this
case, ldo-enable mode should be used and reg_arm/reg_soc/reg_pu should
be swithed to internal ldo instead. Otherwise, no reg_pu regulator
probed successfully by gpu driver, and cause gpu probe failed. Also,
cause cpufreq probe failed too.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qp-sabresd-hdcp.dts |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qp-sabresd-hdcp.dts b/arch/arm/boot/dts/imx6qp-sabresd-hdcp.dts
index 8dce5ca..e19ff13 100644
--- a/arch/arm/boot/dts/imx6qp-sabresd-hdcp.dts
+++ b/arch/arm/boot/dts/imx6qp-sabresd-hdcp.dts
@@ -11,6 +11,11 @@
 
 #include "imx6qp-sabresd.dts"
 
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+};
+
 &hdmi_video {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hdmi_hdcp>;
@@ -20,3 +25,15 @@
 &i2c2 {
 	status = "disable";
 };
+
+&reg_arm {
+	/delete-property/ vin-supply;
+};
+
+&reg_pu {
+	/delete-property/ vin-supply;
+};
+
+&reg_soc {
+	/delete-property/ vin-supply;
+};
-- 
1.7.9.5

